src/http2_connection.erl

```diff
@@ -586,11 +570,11 @@ route_frame({#frame_header{type=?HEADERS}=FH, _Payload},
   when Conn#connection.type == server,
        FH#frame_header.stream_id rem 2 == 0 ->
     go_away(?PROTOCOL_ERROR, Conn);
-route_frame({#frame_header{stream_id=StreamId, type=?HEADERS}=FH, #headers{priority=P}},
-            #connection{}=Conn)
-  when ?IS_FLAG(FH#frame_header.flags, ?FLAG_PRIORITY),
-       P#priority.stream_id == FH#frame_header.stream_id ->
-    rst_stream(StreamId, ?PROTOCOL_ERROR, Conn);
+%route_frame({#frame_header{stream_id=StreamId, type=?HEADERS}=FH, #headers{priority=P}},
```

Comment:
I'll push removing the comment. I moved this logic directly into the `http2_frame_headers`